lib/trino-orc/src/main/java/io/trino/orc/stream/OrcInputStream.java

```diff
@@ -125,19 +127,77 @@ public void readFully(byte[] buffer, int offset, int length)
         }
     }
 
-    public void readFully(Slice buffer, int offset, int length)
+    public void readFully(int[] values, int offset, int length)
             throws IOException
     {
+        if (current == null) {
+            throw new OrcCorruptionException(chunkLoader.getOrcDataSourceId(), "Unexpected end of stream");
+        }
+
         while (length > 0) {
-            if (current != null && current.remaining() == 0) {
+            int remaining = Ints.saturatedCast(current.remaining());
+            if (remaining < Integer.BYTES) {
+                // there might be a value split across the buffers
+                Slice slice = null;
+                if (remaining != 0) {
+                    slice = Slices.allocate(Integer.BYTES);
```

Comment:
Do we need to cache this one-int buffer (in the class, or for the whole method) to avoid potentially reallocating on every boundary? Not sure how often `remaining < Integer.BYTES` is true.

Reply:
I added `tempBuffer8` as a field variable and use it in int[] and long[] methods